File: app/src/main/java/com/schibsted/remotedisplaysample/CustomSimpleOnPageChangeListener.java
package com.schibsted.remotedisplaysample;

import androidx.viewpager.widget.ViewPager;

public class CustomSimpleOnPageChangeListener extends ViewPager.SimpleOnPageChangeListener {
  private OnPageChangePosition onPageChangePosition;

  public CustomSimpleOnPageChangeListener(OnPageChangePosition onPageChangePosition) {
    this.onPageChangePosition = onPageChangePosition;
  }

  @Override
  public void onPageSelected(int position) {
    super.onPageSelected(position);
    onPageChangePosition.onCurrentPageChange(position);
  }

  interface OnPageChangePosition {
    void onCurrentPageChange(int position);
  }
}

File: app/src/main/java/com/schibsted/remotedisplaysample/CustomSimpleOnPageChangeListener.kt
package com.schibsted.remotedisplaysample

import androidx.viewpager.widget.ViewPager

class CustomSimpleOnPageChangeListener(private val onPageChangePosition: OnPageChangePosition) : ViewPager.SimpleOnPageChangeListener() {

    override fun onPageSelected(position: Int) {
        super.onPageSelected(position)
        onPageChangePosition.onCurrentPageChange(position)
    }

    interface OnPageChangePosition {
        fun onCurrentPageChange(position: Int)
    }
}
